hdfs-scheduler/src/main/java/org/apache/mesos/hdfs/state/LiveState.java

```diff
@@ -151,4 +151,50 @@ public boolean apply(String taskId) {
       }
     }).size();
   }
+
+  private class NameNodeState {
+    private String name;
+    private TaskStatus status;
+    private Boolean initialized;
+
+    public NameNodeState(String name) {
+      this.name = name;
+    }
+
+    public Boolean hasStatus() {
+      return status != null;
+    }
+
+    public Boolean initialized() {
+      return status != null && initialized;
+    }
+
+    public void set(TaskStatus status, Boolean initialized) {
+      this.status = status;
+      this.initialized = initialized;
+      log.info(String.format("Set Node '%s' state to status: '%s', initialized: '%s'", name, status, initialized));
+    }
+
+    public void clear(TaskID taskId) {
```

Comment:
let's have this return a boolean true or false if the task was cleared successfully